python/pylibcudf/pylibcudf/tests/io/test_types.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import gc
+
+import pyarrow as pa
+
+import pylibcudf as plc
+
+
+def test_gc_with_table_and_column_input_metadata(monkeypatch):
+    class A(plc.io.types.TableInputMetadata):
+        def __del__(self):
+            print("Deleting A...")
+
+    pa_table = pa.table(
+        {"a": pa.array([1, 2, 3]), "b": pa.array(["a", "b", "c"])}
+    )
+    plc_table = plc.interop.from_arrow(pa_table)
+
+    tbl_meta = A(plc_table)
+
+    gc.disable()
+    gc.collect()
+
+    del tbl_meta
```

Comment:
What are you trying to test here?

If you are tying to test a lifetime issue, I would suggest using `weakref.ref` like:

```python
import weakref

obj_you_want_to_track = 1
test_ref = weakref.ref(obj_you_want_to_track)
del obj_you_want_to_track
assert test_ref() is None
```

Reply:
> What are you trying to test here?

Yeah its a lifetime issue. `ColumnInMetadata` keeps a reference to the `TableInputMeta` object that owns it. This creates a circular reference, so I'm testing that python gc handles this correctly.
 
> If you are tying to test a lifetime issue, I would suggest using `weakref.ref` like:

Okay thanks, will give that a try.

Reply:
Added in https://github.com/rapidsai/cudf/pull/17310/commits/8ae013fd9908aae112fd6a0c5267fd1b6d29b3a1